File: feature/users/src/main/java/com/telynet/telynetusers/feature/users/UserListIntent.java
package com.telynet.telynetusers.feature.users;

public interface UserListIntent {

    class SearchQueryChanged implements UserListIntent {
        private final String query;

        public SearchQueryChanged(String query) {
            this.query = query;
        }

        public String getQuery() {
            return query;
        }
    }

    class FilterVisitedChanged implements UserListIntent {
        private final int filter;

        public FilterVisitedChanged(int filter) {
            this.filter = filter;
        }

        public int getFilter() {
            return filter;
        }
    }

    class OrderByChanged implements UserListIntent {
        private final String orderBy;

        public OrderByChanged(String orderBy) {
            this.orderBy = orderBy;
        }

        public String getOrderBy() {
            return orderBy;
        }
    }
}

File: feature/users/src/main/java/com/telynet/telynetusers/feature/users/UserListIntent.kt
package com.telynet.telynetusers.feature.users

import com.telynet.telynetusers.core.models.entity.User

sealed interface UserListIntent {
    data class SearchQueryChanged(
        val query: String,
    ) : UserListIntent

    data class FilterSelected(
        val filter: VisitFilter,
    ) : UserListIntent

    data class SortSelected(
        val sort: SortOption,
    ) : UserListIntent

    data class ToggleFavorite(
        val user: User,
    ) : UserListIntent
}
